config/db.test.js

```diff
@@ -0,0 +1,35 @@
+import mongoose from 'mongoose';
+import connectDB from './db';
+
+jest.mock("mongoose", () => ({
+    connect: jest.fn(),
+}));
+
+describe('connectDB', () => {
+    let consoleLogSpy;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+    })
+
+    test('connection succesful', async () => {
+        mongoose.connect.mockResolvedValue({
+            connection: { host: 'test-connection' }
+        });
+        await connectDB();
+
+        expect(mongoose.connect).toHaveBeenCalledWith(process.env.MONGO_URL);
+        expect(consoleLogSpy).toHaveBeenCalledWith('Connected To Mongodb Database test-connection'.bgMagenta.white);
+    });
+
+    // NEVER PASS
+    test('connection unsuccessful', async () => {
+        const error = new Error('connection error');
+        mongoose.connect.mockRejectedValue(error);
+        await connectDB();
+
+        expect(consoleLogSpy).toHaveBeenCalledWith('Error in Mongodb: connection error'.bgRed.white);
```

Comment:
Is the reason why this test never pass to be a missing semicolon after 'Mongodb' in the error message?

Reply:
For this, i feel that the expected error should be something like `Error in Mongodb: connection error`. However, because of how the code works in the original file, it would display `Error in Mongodb Error: connection error`, with an additional `Error`.  Let me know what you think about this!

Reply:
Ohhh okay, I think that makes sense! I think the teaching team probably just missed this out haha

However, we can leave it as a bug.